### ISSUE TITLE
10.8.0-775 Open GL: Top or Side Visible Causes Some Textures to Appear All Black or Pink

### ISSUE THREAD

**Issue by DrFlytrap** [vpinball/vpinball#371]
Some items appear as all black or pink unless "Side Visible" or "Top Visible" is unchecked (Pink->Top, Black->Side). This happens in x64 & x86 versions. The black texture issue occurs only in the Open GL version of 10.8.0-775. The pink texture issue occurs in both versions. Neither occurs in 10.7.4.

**Example table where this happens:**
JP's Attack from Mars v4
https://www.vpforums.org/index.php?app=downloads&showfile=11603

**Objects in sample image:**
Layer "pf mesh"
- Wall001 (appears as pink if Top Visible is ENABLED)

Layer "Plastics"
- Wall152, Wall138, and Wall335 (right/top side) have Side Visible DISABLED (for demonstration)
-  Wall283, 289, and Wall333 (left/bottom side) have Side Visible ENABLED (default)

![AFM_TexGlitch](https://user-images.githubusercontent.com/123576665/230186236-6096e5b7-18d5-4cba-9263-5b876f3a71af.jpg)

**Comment by vbousquet** [comment]
I think there are 2 different things here:
- Wall0001 is not meant to be visible (it's a physical part) but is checked as visible. Until 10.8, parts under playfield would be hidden but this has changed since it made things complicated and look bad (for underplayfield, transparency, reflections,...). This is not a 'bug' but more of a change and would need to be changed in the table (either uncheck the visible flag, or give it a full black material)
- the others were bugs that should be solved with the latest (939+) version

Could you have a look and confirm that the bug (second bullet point) is also solved for you ?

![image](https://user-images.githubusercontent.com/2796036/230705326-0bf00c4f-be57-4b9d-9ad2-85c319128342.png)

**Comment by DrFlytrap** [comment]
Thanks so much for the response and for reaching out to JP. I'm not seeing a link to v939 anywhere. Sorry if I'm missing something, but I'm very happy to try it if you could point me in the right direction. I only see 10.8.0-775 on the releases page.

**Comment by DrFlytrap** [comment]
Thanks so much. I tested with v958, and the Side Visible bug does indeed seemed fixed! However...

I encountered a different, strange bug.

The playfield lights in this table are in two layers: the playfield (pf in Image Manager, lights defined on Lights layer), which includes the light inserts with no text, and a separate layer with the text (pf_lights). The text layer seems triggered with Flasher001 (on Flashers) layer.

In one small area in the center of the playfield, the light seems to appear over the text layer when lit, while they appear normal on the rest of the table. Here is an example. The purple lines connect a working example with a similar bugged example:
![AFM_TexGlitch002](https://user-images.githubusercontent.com/123576665/231924515-3f3c443e-6fa7-4b0e-a928-4fdfc71dba75.jpg)

It seems constrained to this one central band of the table. I have compared the settings, XYZ location, etc. of many items, and I can't see anything that may be causing it.
